go-controller/pkg/ovn/controller/services/utils.go

```diff
@@ -338,6 +338,9 @@ func collectServiceVIPs(service *v1.Service) sets.String {
 		}
 		// LoadBalancer
 		for _, ingress := range service.Status.LoadBalancer.Ingress {
+			if ingress.IP == "" {
```

Comment:
nice catch @dcbw 

Is it worth trying to parse this as an IP like we do in the OF code path to ensure the IP is really an IP and not some rando string, or does KAPI take care of this for us?
https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/node/gateway_shared_intf.go#L77

Reply:
@trozet validation takes care of it:

```
func ValidateLoadBalancerStatus(status *core.LoadBalancerStatus, fldPath *field.Path) field.ErrorList {
	allErrs := field.ErrorList{}
	for i, ingress := range status.Ingress {
		idxPath := fldPath.Child("ingress").Index(i)
		if len(ingress.IP) > 0 {
			if isIP := (net.ParseIP(ingress.IP) != nil); !isIP {
				allErrs = append(allErrs, field.Invalid(idxPath.Child("ip"), ingress.IP, "must be a valid IP address"))
			}
		}
		if len(ingress.Hostname) > 0 {
			for _, msg := range validation.IsDNS1123Subdomain(ingress.Hostname) {
				allErrs = append(allErrs, field.Invalid(idxPath.Child("hostname"), ingress.Hostname, msg))
			}
			if isIP := (net.ParseIP(ingress.Hostname) != nil); isIP {
				allErrs = append(allErrs, field.Invalid(idxPath.Child("hostname"), ingress.Hostname, "must be a DNS name, not an IP address"))
			}
		}
	}
	return allErrs
}
```